tests/phpunit/api/v3/ACLPermissionTest.php

```diff
@@ -83,6 +83,22 @@ public function testContactGetNoResultsHook() {
   }
 
   /**
+   * Function tests that an empty where hook returns no results with edit my contact.
+   *
+   * CRM-16512 caused contacts with Edit my contact to be able to view all records.
+   */
+  public function testContactGetNoResultsHookWithEditMyContact() {
+    $this->createLoggedInUser();
+    $this->hookClass->setHook('civicrm_aclWhereClause', array($this, 'aclWhereHookNoResults'));
+    CRM_Core_Config::singleton()->userPermissionClass->permissions = array('access CiviCRM', 'view my contact');
+    $result = $this->callAPISuccess('contact', 'get', array(
+      'check_permissions' => 1,
+      'return' => 'display_name',
+    ));
+    $this->assertEquals(0, $result['count']);
```

Comment:
@eileenmcnaughton so I think ultimately the correct answer to this math problem should be "1" not 0 -- this test creates a contact in the db who has permission to see themself and no one else, so when they fetch contacts with the api there should be 1 result.
That was what my failed patch was trying to achieve. Obviously a different approach would be needed since I was going down the wrong path.
Thoughts?